### ISSUE TITLE
Pid is not working

### ISSUE THREAD

**Issue by holstad87** [Manuel83/cbpi-pidboil#1]
I like the idee of this function. Very nice to have.
Have tested the pid boil and the heating element is on all the time.
Hva used same PID settings as ArduinoPID, so they work fine.

**Comment by Manuel83** [comment]
When is the heating element always on? After above the boil threshold or below?
The current logic always give full throttle above the threshold.

**Comment by holstad87** [comment]
Boil temp is set to 85C . And set the setpoint to 66C. 
Start the test and the element just stay on to full boil. 
Tested this 3 times.

**Comment by Manuel83** [comment]
This one is working right? 

https://github.com/Manuel83/cbpi-PIDArduino

**Comment by holstad87** [comment]
Yes, thet one works perfect

**Comment by Manuel83** [comment]
Ok then it must be some copy past issue. I will have a look now

**Comment by holstad87** [comment]
Thanks :)

**Comment by Manuel83** [comment]
I tried to fix. The issue was missing braces for get_target_temp

Before: `if self.get_target_temp >= float(self.e_boil):`

Now: `if self.get_target_temp() >= float(self.e_boil):`

**Comment by holstad87** [comment]
Nice, i will do some testing tomorrow :)

**Comment by holstad87** [comment]
Working perfect !!!

**Comment by BrewChef** [comment]
Confirming all function works fully.

**Comment by BrauereiAlt** [comment]
Actually, all is not working fully.  As published this add-in functions only in PID mode, unless the Target Temp equals or exceeds Boil Threshold, at which point you get 100% PWM.  To get it to work correctly so that it switches to predefined manual mode at Boil Threshold, I had to change lines 39 & 40 as follows:

            if self.get_temp() >= float(self.e_boil):
                self.heater_on(maxout)

**Comment by BrewChef** [comment]
@BrauereiAlt Could you explain what you meen? I do not understand fully. It seems to work on my system as is, What am I missing?

**Comment by BrauereiAlt** [comment]
I was expecting this:

max output - heater power which is set above boil threshold
Boil Threshold - Above this temperature the heater will be constantly on

to mean that once the process temp reaches the boil threshold temp, the output would switch to max output % PMW.

What is happening instead is like described here:

If the target Temperature is above a configurable threshold the PID will be ignored and heater is                     switched on constantly. This is helpful if you use the same kettle for mashing and boiling.

When process temp is greater than boil threshold temp, the output is on 100% of the time (full power), unless I manually reduce the power output.

**Comment by BrewChef** [comment]
@BrauereiAlt ok, got it, I must admit I have never tried what you describe. I always go 100% pre-boil, and adjust when boil is reached.

**Comment by BrauereiAlt** [comment]
@BrewChef,  what I want is to automate the switch from 100% to my user-definable boil maintenance setting, similar to what the Auber EZ Boil does.  THe difference is that I would have  PID algorithm managing the mash steps.  I added further code to trigger 100%y and the reduce the power at the boil threshold setting.  I need to explore further what was causing it to work correctly in one step but not during a later step, when I had to stop and restart autorun to affect a change in state.
